tests/automated_validation/data_transformation/test_age_groups.py

```diff
@@ -0,0 +1,307 @@
+        ("0_to_8_days", (0, 0.02191780821917808)),
+        ("14_to_17", (14, 17)),
+    ],
+)
+def test_age_group_from_string(string: str, ages: AgeRange) -> None:
+    """Test AgeGroup instantiation from string."""
+    group = AgeGroup.from_string(string)
+    assert group.name == string
+    assert group.start == ages[0]
+    assert np.isclose(group.end, ages[1])
+    assert np.isclose(group.span, ages[1] - ages[0])
+
+
+@pytest.mark.parametrize(
+    "string, match",
+    [
+        ("invalid_format", "Invalid age group name format:"),
+        (
+            "0_to_5_invalid_unit",
+            "Invalid unit: invalid_unit. Must be 'days', 'months', or 'years'.",
+        ),
+    ],
+)
+def test_age_group_invalid_string(string: str, match: str) -> None:
+    """Test AgeGroup instantiation from invalid string."""
+    with pytest.raises(ValueError, match=match):
+        AgeGroup.from_string(string)
+
+
+def test_age_group_from_range() -> None:
+    """Test AgeGroup instantiation from range."""
+    group = AgeGroup.from_range(0, 5)
+    assert group.name == "0_to_5"
+    assert group.start == 0
+    assert group.end == 5
+    assert group.span == 5
+
+
+@pytest.mark.parametrize(
+    "group_name, group_ages, fraction",
+    [
+        ("0_to_5_years", (0, 5), 1.0),
+        ("0_to_10_years", (0, 10), 1.0),
+        ("3_to_8_years", (3, 8), 2 / 5),
+        ("6_to_10_years", (6, 10), 0.0),
+    ],
+)
+def test_age_group_fraction_contained_by(
+    group_name: str, group_ages: AgeRange, fraction: float
+) -> None:
+    """Test that we get the correct amount of overlap between two age groups."""
+    group = AgeGroup("0_to_5_years", 0, 5)
+
+    other_group = AgeGroup(group_name, *group_ages)
+    assert group.fraction_contained_by(other_group) == fraction
+
+
+def check_example_age_schema(age_schema: AgeSchema) -> None:
+    """Check that the example age schema was instantiated correctly, regardless of method."""
+    assert len(age_schema) == 3
+    assert age_schema[0] == AgeGroup("0_to_5", 0, 5)
+    assert age_schema[1] == AgeGroup("5_to_10", 5, 10)
+    assert age_schema[2] == AgeGroup("10_to_15", 10, 15)
+    assert age_schema.range == (0, 15)
+    assert age_schema.span == 15
+
+
+def test_age_schema_instantiation(
+    sample_age_tuples: list[AgeTuple],
+    sample_df_with_ages: pd.DataFrame,
+) -> None:
+    """Test the AgeSchema class instantiation."""
+    for age_schema in [
+        AgeSchema.from_tuples(sample_age_tuples),
+        AgeSchema.from_ranges([(tuple[1], tuple[2]) for tuple in sample_age_tuples]),
+        AgeSchema.from_strings([tuple[0] for tuple in sample_age_tuples]),
+        AgeSchema.from_dataframe(sample_df_with_ages),
+    ]:
+        check_example_age_schema(age_schema)
+
+
+@pytest.mark.parametrize(
+    "age_groups, err_match",
+    [
+        ([("0_to_5", 0, 5), ("4_to_10", 4, 10)], "Overlapping age groups"),
+        ([("0_to_5", 0, 5), ("6_to_10", 6, 10)], "Gap between consecutive age groups"),
```

Comment:
Not sure if I misread the function earlier but you are checking for this somewhere so you can ignore the other comment.